packages/legend-query/src/stores/QueryBuilderPropertySearchPanelState.ts

```diff
@@ -0,0 +1,398 @@
+            node.type.name === PRIMITIVE_TYPE.STRICTTIME ||
+            node.type.name === PRIMITIVE_TYPE.LATESTDATE)
+        ) {
+          return false;
+        }
+        return true;
+      }),
+    );
+  }
+
+  removeBooleanPropertyNodes(): void {
+    this.setBooleanMappedPropertyNodes(
+      this.searchedMappedPropertyNodes.filter((node) => {
+        if (
+          node.type instanceof PrimitiveType &&
+          node.type.name === PRIMITIVE_TYPE.BOOLEAN
+        ) {
+          return false;
+        }
+        return true;
+      }),
+    );
+  }
+
+  updateOneManyRowsIncludedMappedProperties(): void {
+    this.isOneManyRowsIncluded
+      ? this.setOneManyMappedPropertyNodes(this.searchedMappedPropertyNodes)
+      : this.removeOneManyPropertyNodes();
+  }
+
+  updateClassIncludedMappedProperties(): void {
+    this.isClassIncluded
+      ? this.setClassMappedPropertyNodes(this.searchedMappedPropertyNodes)
+      : this.removeClassPropertyNodes();
+  }
+
+  updateStringIncludedMappedProperties(): void {
+    this.isStringIncluded
+      ? this.setStringMappedPropertyNodes(this.searchedMappedPropertyNodes)
+      : this.removeStringPropertyNodes();
+  }
+
+  updateBooleanIncludedMappedProperties(): void {
+    this.isBooleanIncluded
+      ? this.setBooleanMappedPropertyNodes(this.searchedMappedPropertyNodes)
+      : this.removeBooleanPropertyNodes();
+  }
+
+  updateNumberIncludedMappedProperties(): void {
+    this.isNumberIncluded
+      ? this.setNumberMappedPropertyNodes(this.searchedMappedPropertyNodes)
+      : this.removeNumberPropertyNodes();
+  }
+
+  updateDateIncludedMappedProperties(): void {
+    this.isDateIncluded
+      ? this.setDateMappedPropertyNodes(this.searchedMappedPropertyNodes)
+      : this.removeDatePropertyNodes();
+  }
+
+  get filteredPropertyNodes(): QueryBuilderExplorerTreeNodeData[] {
+    return this.oneManyMappedPropertyNodes.filter(
+      (p) =>
+        this.classMappedPropertyNodes.includes(p) &&
+        this.stringMappedPropertyNodes.includes(p) &&
+        this.numberMappedPropertyNodes.includes(p) &&
+        this.booleanMappedPropertyNodes.includes(p) &&
+        this.dateMappedPropertyNodes.includes(p),
+    );
+  }
+
+  fetchAllPropertyNodes(): void {
+    const treeData = this.queryBuilderState.explorerState.treeData;
+    let currentLevelPropertyNodes: QueryBuilderExplorerTreeNodeData[] = [];
+    let nextLevelPropertyNodes: QueryBuilderExplorerTreeNodeData[] = [];
+    if (treeData?.nodes.values()) {
+      Array.from(treeData.nodes.values())
+        .slice(1)
+        .forEach((node) => {
+          if (node.mappingData.mapped && !node.isPartOfDerivedPropertyBranch) {
+            currentLevelPropertyNodes.push(node);
+            this.allMappedPropertyNodes.push(node);
+          }
+        });
+    }
+    let currentDepth = 1;
+    const maxDepth = 15;
+    while (currentLevelPropertyNodes.length && currentDepth <= maxDepth) {
```

Comment:
Do we handle loops in here. Like Firm -> Person -> Firm?

Reply:
No, we don't handle loops in here. I think we should fix handling of loops in `mappingAlgo`